### PR TITLE
chore(rln_keystore_generator): generate and persist credentials

### PR DESCRIPTION
**Pull request by rymnc**
# Description

A follow up PR to #1925

# Changes

- [x] Creates an instance of the rln object
- [x] Generates credentials
- [x] Uses dummy chainId, contractAddress and treeIndex to write to a keystore

The next follow up PR will actually execute the registration if `--execute` is true

### CODE REVIEW

File: tools/rln_keystore_generator/external_config.nim [hunk 1]
```diff
@@ -51,14 +51,20 @@ type
...
       name: "rln-relay-cred-password" }: string
 
 proc loadConfig*(T: type RlnKeystoreGeneratorConf): Result[T, string] =
   try:
     let conf = RlnKeystoreGeneratorConf.load()
+    if conf.rlnRelayCredPath == "":
+      return err("--rln-relay-cred-path must be set")
+    if conf.rlnRelayEthContractAddress == "":
+      return err("--rln-relay-eth-contract-address must be set")
+    if conf.rlnRelayCredPassword == "":
```
Comment (Ivansete-status) [inline tools/rln_keystore_generator/external_config.nim:66]: Shall we also ask for a minimum password strength? ( e.g. minimum length, a-zA-Z, alphanumeric symbols, ... )

File: tools/rln_keystore_generator/rln_keystore_generator.nim [hunk 2]
```diff
@@ -24,7 +29,48 @@ when isMainModule:
   
   debug "configuration", conf = $conf
 
-  # initialize keystore
+  # 2. initialize rlnInstance
+  let rlnInstanceRes = createRLNInstance(d=20, 
+                                         tree_path = genTempPath("rln_tree", "rln_keystore_generator"))
+  if rlnInstanceRes.isErr():
+    error "failure while creating RLN instance", error=rlnInstanceRes.error()
```
Comment (Ivansete-status) [inline tools/rln_keystore_generator/rln_keystore_generator.nim:36]: I think we tend not to use `()` for "name" attributes, although is being used in many places in this file
```suggestion
    error "failure while creating RLN instance", error=rlnInstanceRes.error
```

File: tools/rln_keystore_generator/rln_keystore_generator.nim [hunk 3]
```diff
@@ -24,7 +29,48 @@ when isMainModule:
...
+    membershipGroups: @[MembershipGroup(
+      membershipContract: MembershipContract(
+        chainId: "1155511",
+        address: conf.rlnRelayEthContractAddress,
+      ),
+      treeIndex: 0,
+    )]
+  )
+
+  let persistRes = addMembershipCredentials(conf.rlnRelayCredPath, @[keystoreCred], conf.rlnRelayCredPassword, RLNAppInfo)
```
Comment (Ivansete-status) [inline tools/rln_keystore_generator/rln_keystore_generator.nim:67]: Better try to have the code as vertical as possible and each line less than 80 characters :)
```suggestion
  let persistRes = addMembershipCredentials(conf.rlnRelayCredPath,
                                                                         @[keystoreCred],
                                                                         conf.rlnRelayCredPassword,
                                                                         RLNAppInfo)
```

File: tools/rln_keystore_generator/rln_keystore_generator.nim [hunk 4]
```diff
@@ -24,7 +29,48 @@ when isMainModule:
...
+                       idCommitment = credential.idCommitment.inHex()
+
+  # 4. write to keystore
+  ## TODO: after hooking up to the OnchainGroupManager, 
+  ## obtain chainId and treeIndex from the contract
+  let keystoreCred = MembershipCredentials(
+    identityCredential: credential,
+    membershipGroups: @[MembershipGroup(
+      membershipContract: MembershipContract(
+        chainId: "1155511",
```
Comment (Ivansete-status) [inline tools/rln_keystore_generator/rln_keystore_generator.nim:60]: Shall we have a comment on this value? Can this come from config as well?

**Review by Ivansete-status — APPROVED** [review]
LGTM as far as I could understand :P!
Just added minor comments

File: tools/rln_keystore_generator/rln_keystore_generator.nim (hunk 4 above)
Comment (rymnc) [reply tools/rln_keystore_generator/rln_keystore_generator.nim:60]: This is a dummy value that will be replaced with the next iteration!

File: tools/rln_keystore_generator/external_config.nim (hunk 1 above)
Comment (rymnc) [reply tools/rln_keystore_generator/external_config.nim:66]: can I cover this in a future iteration? tracking issue for it here - https://github.com/waku-org/nwaku/issues/1930

File: tools/rln_keystore_generator/rln_keystore_generator.nim (hunk 2 above)
Comment (rymnc) [reply tools/rln_keystore_generator/rln_keystore_generator.nim:36]: Addressed in fed49b55

File: tools/rln_keystore_generator/rln_keystore_generator.nim (hunk 3 above)
Comment (rymnc) [reply tools/rln_keystore_generator/rln_keystore_generator.nim]: Addressed in 131519d7

File: tools/rln_keystore_generator/external_config.nim (hunk 1 above)
Comment (Ivansete-status) [reply tools/rln_keystore_generator/external_config.nim:66]: Yes of course:) Thanks!

Comment (alrevuelta) [reply tools/rln_keystore_generator/external_config.nim:66]: unsure if we can do that. is the keystore taken from some spec (eg EIP)? if so, if the EIP does not enforce any password strength, then we can enforce it on top.

eg, not the one we are using but https://eips.ethereum.org/EIPS/eip-2335

> The password is a string of arbitrary unicode characters. The password is first converted to its NFKD representation, then the control codes (specified below) are stripped from the password and finally it is UTF-8 encoded.

there are no requirements beyond that

**Review by alrevuelta — APPROVED** [review]

**Pull request merged**